diff --git a/node_modules/@a2a-js/sdk/dist/server/index.cjs b/node_modules/@a2a-js/sdk/dist/server/index.cjs
index a127620..fa5fbec 100644
--- a/node_modules/@a2a-js/sdk/dist/server/index.cjs
+++ b/node_modules/@a2a-js/sdk/dist/server/index.cjs
@@ -643,6 +643,17 @@ var DefaultPushNotificationSender = class {
       if (pushConfig.token) {
         headers[this.options.tokenHeaderName] = pushConfig.token;
       }
+      console.log("received pushConfig.authentication:", pushConfig.authentication);
+      if (pushConfig.authentication && pushConfig.authentication.credentials) {
+        const { schemes, credentials } = pushConfig.authentication;
+
+        if (schemes.includes("Bearer")) {
+          headers["Authorization"] = `Bearer ${credentials}`;
+        }
+        if (schemes.includes("Basic")) {
+          headers["Authorization"] = `Basic ${credentials}`;
+        }
+      }
       const response = await fetch(url, {
         method: "POST",
         headers,
